src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,139 @@
+        try {
+            createFile();
+            FileWriter fileWriter = new FileWriter(textFile);
+            for (Task t : taskList) {
+                if (t instanceof ToDo) {
+                    fileWriter.write("T | ");
+                } else if (t instanceof Deadline) {
+                    fileWriter.write("D | ");
+                } else if (t instanceof Event) {
+                    fileWriter.write("E | ");
+                }
+                if (t.isComplete()) {
+                    fileWriter.write("1 | " + t.taskName);
+                } else {
+                    fileWriter.write("0 | " + t.taskName);
+                }
+                if (t instanceof Deadline) {
+                    fileWriter.write(" | " + ((Deadline) t).getDate());
+                }
+                if (t instanceof Event) {
+                    fileWriter.write(" | " + ((Event) t).getDate());
+                }
+                fileWriter.write("\n");
+            }
+            fileWriter.close();
+        } catch (IOException exception) {
+            System.out.println(exception);
+        }
+    }
+
+    /**
+     * Method to read an existing text file and convert it into a corresponding task list within
+     * the Dukebot.
+     *
+     * @param file the text file in which the task list is created from
+     * @param taskList the corresponding task list to copy the text file over
+     * @throws IOException whenever an issue with reading the file arises
+     */
+    public static void convert(File file, ArrayList<Task> taskList) throws IOException {
+        Scanner scan = new Scanner(file);
+        while (scan.hasNext()) {
+            String[] parsed = splitter(scan.nextLine());
+            String type = parsed[0];
+            boolean isDone = parsed[1].equals("1");
+            String taskName;
+            if (type.equals("T")) {
+                taskName = parsed[2].substring(1);
+                taskList.add(new ToDo(taskName, isDone));
+            } else {
+                taskName = parsed[2].substring(1);
+                LocalDate date = dateConverter(parsed[3].substring(1));
+                if (type.equals("D")) {
+                    taskList.add(new Deadline(taskName, isDone, date));
+                } else if (type.equals("E")) {
+                    taskList.add(new Event(taskName, isDone, date));
+                }
+            }
+        }
+        scan.close();
+    }
+
+    private static String[] splitter(String input) {
+        String[] result = new String[4];
+        int counter = 0;
+        result[0] = "";
+        for (int i = 0; i < input.length(); i++) {
+            if (input.charAt(i) == '|') {
+                counter++;
+                result[counter] = "";
+            } else if (input.charAt(i) != ' ' || counter >= 2) {
+                result[counter] += input.charAt(i);
+            }
+        }
+        return result;
+    }
+
+    private static LocalDate dateConverter(String date) {
```

Comment:
Is this necessary? What if users choose to input dates in another format? So, I would think it is better to make the user input a certain format like yyyy-mm-dd instead of catering to the different formats that a user can input.